temporian/implementation/numpy/data/dtype_normalization.py

```diff
@@ -236,3 +238,103 @@ def normalize_index_key_list(
     normalized_indexes = [normalize_index_key(x) for x in normalized_indexes]
 
     return normalized_indexes
+
+
+def normalize_dtype(x: Any) -> DType:
```

Comment:
What about using this map (https://github.com/google/temporian/blob/main/temporian/core/data/dtype.py#L91), and possibly utility function (if we have one)?